Add shadowrootslotassignment attribute (#43753)
mdn/content: files/en-us/web/api/shadowroot/slotassignment/index.md

Reviewer: Two ways to do this now right? Something like ...

```suggestion
The read-only **`slotAssignment`** property of the {{domxref("ShadowRoot")}} interface returns the _slot assignment mode_ for the shadow DOM tree. Nodes are either automatically assigned (`named`) or manually assigned (`manual`).
The value of this property is defined using the `slotAssignment` option when calling {{domxref("Element.attachShadow()")}}, or using the [`shadowrootslotassignment` attribute](/en-US/docs/Web/HTML/Reference/Elements/template#shadowrootslotassignment) on {{htmleelement("template")}} when declaratively creating a shadow root.
```

{{APIRef("Shadow DOM")}}

The read-only **`slotAssignment`** property of the {{domxref("ShadowRoot")}} interface returns the _slot assignment mode_ for the shadow DOM tree. Nodes are either automatically assigned based on name matching (`named`) or manually assigned (`manual`).

> The value of this property is defined using the `slotAssignment` option when calling {{domxref("Element.attachShadow()")}}, or using the [`shadowrootslotassignment`](/en-US/docs/Web/HTML/Reference/Elements/template#shadowrootslotassignment) attribute on {{htmlelement("template")}} when declaratively creating a shadow root.

Reviewer 2: ```suggestion
The value of this property is defined using the `slotAssignment` option when calling {{domxref("Element.attachShadow()")}}, or using the [`shadowrootslotassignment`](/en-US/docs/Web/HTML/Reference/Elements/template#shadowrootslotassignment) attribute on a {{htmlelement("template")}} element when declaratively creating a shadow root.
```

## Value

A string that can be one of:

- `named`
  - : Elements are automatically assigned to {{HTMLElement("slot")}} elements within this shadow root.

>     Any descendants of the host with a `slot` attribute that matches the `name` attribute of a `<slot>` within this shadow root will be assigned to that slot.

Author: Again here, "top-level children" might be better.

Reviewer: ```suggestion
    Any top-level children of the host with a `slot` attribute that matches the `name` attribute of a `<slot>` within this shadow root will be assigned to that slot.
```

    Any top-level children of the host with no `slot` attribute will be assigned to a `<slot>` with no `name` attribute (the "default slot"), if one is present.
- `manual`
  - : Elements are not automatically assigned to {{HTMLElement("slot")}} elements.
    Instead, they must be manually assigned with {{domxref("HTMLSlotElement.assign()")}}.

## Examples

### Basic usage

In the example below, the `assign()` method is used to display the correct tab in a tabbed application.
The function is called and passed the panel to show, which is then assigned to the slot.

> We test if the `slotAssignment` is `named` to prevent an exception from being raised when {{domxref("HTMLSlotElement.assign()")}} is called.

Reviewer 2: I won't block on this, but it would be great to see a live example to show what this does.

Reviewer: It's not very interesting - generally all you can do with this one is check if manual assignment is supported, and if it is, whether it is used.

This is a trivial example of use, which is really just logging that you have the expected kind of assignment. I didn't write it, just added words.

So it would be OK, but I'm more tempted to reduce this example.

Reviewer 2: OK, that's fair.
